src/BatteryStateOfChargeService.cpp

```diff
@@ -16,21 +20,51 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return BATTERY_AMP_HOUR_CAPACITY-ampHours_ ;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+    if(previousCurrent_<0)
+    {
     return false;
```

Comment:
Stuff inside a function/statement (In this case , the if statement) should be 4 spaces in from the bracket.